### PR TITLE
Exclude files from backup

### LINKED ISSUE #537: Files are included in iTunes/iCloud backup
Hi,

We found out that the analytics files are saved in the Library/Application Support folder. Files in this folder are included in the iTunes/iCloud backup, unless marked as 'exclude from backup'.

This can be a reason for rejection by Apple (if the files become large), so I hope you can change this.

There are two solutions:

1. Mark folder and files with NSURLIsExcludedFromBackupKey. I.e. use this: `BOOL success = [URL setResourceValue: [NSNumber numberWithBool: YES] forKey: NSURLIsExcludedFromBackupKey error: &error];`
2. Store all files in Library/Caches.

More info: https://developer.apple.com/library/ios/qa/qa1719/_index.html

Cheers!
Peter

### PR DESCRIPTION
**Pull request by f2prateek**
Closes #537

### CODE REVIEW

**Comment by f2prateek** [comment]
Reference: https://developer.apple.com/library/ios/qa/qa1719/_index.html

**Comment by f2prateek** [comment]
cc @wcjohnson11 @tonyxiao @ladanazita for review

**Comment by f2prateek** [comment]
and @RetepV as well!

**Comment by wcjohnson11** [comment]
lgtm! 👍

File: Analytics/Classes/Internal/SEGSegmentIntegration.m [hunk 1]
```diff
@@ -102,6 +102,9 @@ - (id)initWithAnalytics:(SEGAnalytics *)analytics
         dispatch_sync(dispatch_get_main_queue(), ^{
             self.flushTimer = [NSTimer scheduledTimerWithTimeInterval:30.0 target:self selector:@selector(flush) userInfo:nil repeats:YES];
         });
+        [self addSkipBackupAttributeToItemAtPath:self.userIDURL];
```
Comment (tonyxiao) [inline Analytics/Classes/Internal/SEGSegmentIntegration.m]: what about `anonymousIdURL`?

Comment (f2prateek) [inline Analytics/Classes/Internal/SEGSegmentIntegration.m]: oh good catch, I just copied the ones here https://github.com/segmentio/analytics-ios/blob/46a56f6fe0cd6866f3b6fac370d7815fc341aad0/Analytics/Classes/Internal/SEGSegmentIntegration.m#L498-L500 but look we're missing anonymousID there

**Pull request merged**